kyaml/yaml/util.go

```diff
@@ -47,3 +47,22 @@ func DeriveSeqIndentStyle(originalYAML string) string {
 
 	return string(CompactSequenceStyle)
 }
+
+// keyLineBeforeSeqElem iterates through the lines before the first seqElement
+// and tries to find the non-comment key line for the sequence node
+func keyLineBeforeSeqElem(lines []string, seqElemIndex int) string {
+	// start with the previous line of sequence element
+	i := seqElemIndex - 1
+	for i >= 0 {
+		// split the line into 2 parts, non-comment and comment
+		// SplitN always ensure that the result array is at least of size 1
+		keyLine := strings.SplitN(lines[i], "#", 2)[0]
+		// check if the non-comment is not just spaces
+		if len(strings.Trim(keyLine, " ")) != 0 {
+			return keyLine
+		}
+		// keep going up, till we find a non-comment line
+		i--
+	}
+	return ""
```

Comment:
I don't know I found the above code to be clever, so made an attempt an alternative version:

```
i := seqElemIndex -1
for line := lines[i]; i >= 0; i-- {
      trimmedLine := strings.Trim(line, " ")
      if strings.HasPrefix(trimmedLine, "#") { // commented line
         continue
      }
      // we have a non commented line which can have a trailing comment
    parts = strings.SplitN(line, "#", 2)
    return parts[0] // throw away the trailing comment part
}
return ""
```